src/driver/opam.ml

```diff
@@ -30,18 +30,31 @@ let prefix () =
       prefix := Some p;
       p
 
-let deps_of_opam_result line =
-  match Astring.String.fields ~empty:false line with
-  | [ name; version ] -> [ { name; version } ]
-  | _ -> []
-
-let all_opam_packages () =
-  Util.lines_of_process
-    Cmd.(
-      opam % "list" % "--switch" % get_switch () % "--columns=name,version"
-      % "--color=never" % "--short")
-  |> List.map deps_of_opam_result
-  |> List.flatten
+let all_opam_packages =
+  let result = ref None in
+  fun () ->
+    match !result with
```

Comment:
It took me a minute to realize this was memoization, could we define an auxiliary `memo` function to abstract that and make the function easier to read?

Reply:
Sure, there are a couple of others there too.